### PR TITLE
PSR-109

### PR DESCRIPTION
**Pull request by dephillipsmichael**
Created a new type of step that acts like a multi-choice selection step but displays in a collection view with each collection view cell operating like a tableview selection cell.  Switched all the body part selection steps to use this on tasks psoriasis draw, joint counting, and joint swelling.

@syoung-smallwisdom I wanted to get this on your radar.  Feedback is welcome although unless it is a critical mistake I will probably hold off on fixing until we tackle a generic solution.   Some of the more applicable components are that I got a navigation header working in as a collection view header.  I also was able to tweak a flow layout for the collection view to display a grid with X number of columns, or in this case 2.

### CODE REVIEW

File: Psorcast/PsorcastValidation/Assets.xcassets/SelectionHands.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "images" : [
```
Comment (syoung-smallwisdom) [inline Psorcast/PsorcastValidation/Assets.xcassets/SelectionHands.imageset/Contents.json]: Be sure to run this on device and also make sure that testers check multiple devices b/c of the image caching issue.

File: Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    @IBOutlet public var imageView: UIImageView?
+    
+    open override var isSelected: Bool {
+        didSet {
+            updateColorsAndFonts()
+        }
+    }
+    
+    open private(set) var titleTextType: RSDDesignSystem.TextType = .small
+    open private(set) var detailTextType: RSDDesignSystem.TextType = .bodyDetail
```
Comment (syoung-smallwisdom) [inline Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift]: I think that these are both dynamic. Have you checked what happens when the user has the text size set to super big?

File: Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    func updateColorsAndFonts() {
+        let designSystem = self.designSystem ?? RSDDesignSystem()
+        let background = self.backgroundColorTile ?? RSDGrayScale().white
+        let contentTile = designSystem.colorRules.tableCellBackground(on: background, isSelected: isSelected)
+        
+        contentView.backgroundColor = contentTile.color
+        titleLabel?.textColor = designSystem.colorRules.textColor(on: contentTile, for: titleTextType)
+        titleLabel?.font = designSystem.fontRules.font(for: titleTextType, compatibleWith: traitCollection)
+        detailLabel?.textColor = designSystem.colorRules.textColor(on: contentTile, for: detailTextType)
+        detailLabel?.font = designSystem.fontRules.font(for: detailTextType, compatibleWith: traitCollection)
```
Comment (syoung-smallwisdom) [inline Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift]: If you want to make sure that your text will fit (non-dynamic) you can also use the `baseFont()` method here.

File: Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift (hunk 2 above)
Comment (dephillipsmichael) [reply Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift]: I have not, how do I check that?

Comment (syoung-smallwisdom) [reply Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift]: oh, just through the settings app. `Settings -> General -> Accessibility -> Larger Text`

Turn it on and then use the slider to set it to *really* big. The fonts returned respect the dynamic font sizing *if* they are "dynamic" fonts (verse static).

Comment (dephillipsmichael) [reply Psorcast/PsorcastValidation/RSDSelectionCollectionViewCell.swift]: That's a great debugging tip, thanks!

**Pull request merged**